File: src/test/java/TestMultiplication.java
import org.junit.Assert;
import org.junit.Test;


public class TestMultiplication {

    @Test
    public void testSimple() {
        IFraction fraction = TestUtils.createFraction(2, 3);
        IFraction result = fraction.times(new NotImplementedFraction(5, 7));

        Assert.assertEquals((Integer) 10, result.getNumerator());
        Assert.assertEquals((Integer) 21, result.getDenominator());
    }

    @Test
    public void testNormalisation() {
        IFraction fraction = TestUtils.createFraction(2, 8);
        IFraction result = fraction.times(new NotImplementedFraction(4, 2));

        Assert.assertEquals((Integer) 1, result.getNumerator());
        Assert.assertEquals((Integer) 2, result.getDenominator());
    }
}

File: src/test/kotlin/TestMultiplication.kt
import org.junit.Assert
import org.junit.Test

class TestMultiplication {
    @Test
    fun testSimple() {
        val fraction = TestUtils.createFraction(2, 3)
        val result = fraction.times(NotImplementedFraction(5, 7))

        Assert.assertEquals(10, result.numerator)
        Assert.assertEquals(21, result.denominator)
    }

    @Test
    fun testNormalisation() {
        val fraction = TestUtils.createFraction(2, 8)
        val result = fraction.times(NotImplementedFraction(4, 2))

        Assert.assertEquals(1, result.numerator)
        Assert.assertEquals(2, result.denominator)
    }
}
